SkietbaanBE/SkietbaanBE/Controllers/UserController.cs

```diff
@@ -10,50 +14,70 @@ namespace SkietbaanBE.Controllers
     public class UserController : Controller
     {
         private ModelsContext _context;
-        public UserController(ModelsContext db) {
+        public UserController(ModelsContext db)
+        {
             _context = db;
         }
 
         //public UserController() { }
 
         // GET: api/User
         [HttpGet]
-        public IEnumerable<User> Get() {
+        public IEnumerable<User> Get()
+        {
             var users = _context.Users.ToArray<User>();
             return users;
         }
 
         // GET: api/User/5
         [HttpGet("{id}", Name = "Get")]
-        public User Get(int id) {
-            User user = _context.Users.Find(id);
+        public async Task<User> Get(int id)
+        {
+            User user = await _context.Users.FindAsync(id);
             return user;
         }
 
-        [HttpGet("test/{ret}", Name = "Test")]
-        public string Test(string ret) {
-            return $"Returned : {ret}";
-        }
-
         // POST: api/User
         [HttpPost]
-        public virtual void Post([FromBody] User user) {
-            _context.Add(user);
-            _context.SaveChanges();
+        public async Task<HttpResponseMessage> Post([FromBody] User user)
+        {
+            
+            if (ModelState.IsValid)
+            {
+                //get today's date and save it under user entry date
+                DateTime currentDate = new DateTime();
+                user.EntryDate = currentDate;
```

Comment:
Why is currentDate a new variable? Just assign it directly

Reply:
getting today's date

Reply:
`user.EntryDate = new DateTime(); `
Same effect, no surplus of variables